odoo/addons/base/ir/ir_ui_view.py

```diff
@@ -328,13 +328,22 @@ def _check_xml(self):
                     view_docs = view_docs[0]
                 validator = self._relaxng()
                 for view_arch in view_docs:
-                    version = view_arch.get('version', '7.0')
-                    if parse_version(version) < parse_version('7.0') and validator and not validator.validate(view_arch):
-                        for error in validator.error_log:
-                            _logger.error(tools.ustr(error))
-                        raise ValidationError(_('Invalid view definition'))
-                    if not valid_view(view_arch):
-                        raise ValidationError(_('Invalid view definition'))
+                    if view.type in ['form', 'kanban']:
+                        if not valid_view(view_arch):
+                            raise ValidationError(_('Invalid view definition'))
+                    if view.type in ['search', 'calendar', 'gantt', 'tree', 'graph', 'pivot']:
```

Comment:
That's *really* not correct. `valid_view` applies to every view type (including views which are not even in core), not just form and kanban.

A *much* better alternative would be to e.g. move the rng-based view validation over to `view_validation.py`. And a further improvement would be to *split* the big universal RNG into view-specific schemas (possibly with e.g. [modular mixins](http://relaxng.org/tutorial-20011203.html#IDAX4YR) for the common bits).